### PR TITLE
Chore/ts --isolatedModules issue

### PR DESCRIPTION
**Pull request by zakharchenkoAndrii**
# Why
We have `--isolatedModules` flag enabled in our TS config, after upgrading to expo-47 our typescript linter started to complain on a few directories within the expo.

# How
Satisfied TS by exporting types separately.

# Test Plan

enable `--isolatedModules` without the fix and see the error. 

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-location/CHANGELOG.md [hunk 1]
```diff
@@ -7,7 +7,7 @@
 ### 🎉 New features
 
 ### 🐛 Bug fixes
-
+- Fixed `Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.` ts issue ([#20239](https://github.com/expo/expo/pull/20239) by [@zakharchenkoAndrii](https://github.com/zakharchenkoAndrii))
```
Comment (lukmccall) [inline packages/expo-location/CHANGELOG.md:10]:
```suggestion
- Fixed `Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.` ts issue. ([#20239](https://github.com/expo/expo/pull/20239) by [@zakharchenkoAndrii](https://github.com/zakharchenkoAndrii))

```

File: packages/expo-sensors/CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,7 @@
 ### 🎉 New features
 
 ### 🐛 Bug fixes
+- Fixed `Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.` ts issue ([#20239](https://github.com/expo/expo/pull/20239) by [@zakharchenkoAndrii](https://github.com/zakharchenkoAndrii))
```
Comment (lukmccall) [inline packages/expo-sensors/CHANGELOG.md:10]:
```suggestion

- Fixed `Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.` ts issue. ([#20239](https://github.com/expo/expo/pull/20239) by [@zakharchenkoAndrii](https://github.com/zakharchenkoAndrii))
```

**Review by lukmccall — APPROVED** [review]

**Comment by lukmccall** [comment]
Please, rebuild the js of edited packages using `yarn build` in the package directory.

**Comment by zakharchenkoAndrii** [comment]
> Please, rebuild the js of edited packages using `yarn build` in the package directory.

done :)

File: packages/expo-sensors/CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,8 @@
 
 ### 🐛 Bug fixes
 
+- Fixed `Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.` ts issue ([#20239](https://github.com/expo/expo/pull/20239) by [@zakharchenkoAndrii](https://github.com/zakharchenkoAndrii))
```
Comment (Simek) [inline packages/expo-sensors/CHANGELOG.md:11]:
```suggestion
- Export types with type-only annotation to fix build when using `isolatedModules` flag.  ([#20239](https://github.com/expo/expo/pull/20239) by [@zakharchenkoAndrii](https://github.com/zakharchenkoAndrii))
```

Comment (zakharchenkoAndrii) [reply packages/expo-sensors/CHANGELOG.md]: done

**Pull request merged**